### DISCUSSION TITLE
Nuxt 3 Health Check?

### DISCUSSION THREAD

**Discussion by justindasilva** [nuxt/nuxt#17493] [Questions]
What would be the best way to handle a `/healthcheck` endpoint in Nuxt 3?

I'm curious about the case where the app is actually broken in which case the endpoint would throw a 404.

**Comment by pi0** [comment]
Hi. You can actually create a simple one confirming the server is working:

```js
// server/routes/healthcheck.ts

export default () => true
```

This simply confirms the main server is up when calling endpoint.

**Comment by justindasilva** [comment]
So I guess it's up to the caller of the healthcheck endpoint to determine if a response is actually coming back in the first place? As in, the caller would determine if the request to the `/healthcheck` endpoint returns a good response or a 404.
I'm just curious if there is any way that the `/healthcheck` endpoint can return some default response (error) if the app was down. If the server was apache, I feel like it would be possible to set up an HTAccess rule to do this. This way, you can gaurentee that if you hit `/healthcheck`, no matter what state the app is in, you would get back some response. (Again, just curious. Perfectly okay with the caller having the logic in place to deal with it.)

Also, after doing a bit of research online, I ended up with the following: (Re: https://blog.logrocket.com/how-to-implement-a-health-check-in-node-js/)

```
import { defineEventHandler } from '#server-utils'

export default defineEventHandler(event => {

  let healthcheck

  try {
    healthcheck = {
      uptime: process.uptime(),
      responsetime: process.hrtime(),
      message: `OK`,
      timestamp: Date.now(),
    }
  } catch (error) {
    healthcheck.message = error.message
    event.res.statusCode = 503
  }

  return healthcheck
})
```

Curious, what other ideas people might have for a good health check.

**Reply by wobsoriano** [reply]
Add process.memoryUsage too 😄

**Comment by wobsoriano** [comment]
You can use this https://www.npmjs.com/package/server-health. It's basically

```ts
import pkg from '~/package.json'

const startTime = new Date();

export default defineEventHandler(() => {
  return {
    status: 'success', // update to fail if fail
    uptime: process.uptime(),
    upSince: startTime,
    localTime: new Date(),
    service: {
      name: pkg.name,
      description: pkg.description,
      version: pkg.version,
    },
    env: {
      nodeEnv: process.env.NODE_ENV,
      nodeVersion: process.version,
      processName: process.title,
      pid: process.pid,
      cwd: process.cwd(),
    },
  }
})
```
